### ISSUE TITLE
"set severity-output number" ignored when set via `master` socket

### ISSUE THREAD

**Issue by shadyabhi** [haproxy/haproxy#2350]
### Detailed Description of the Problem

There are two ways to interact with HAproxy socket, via worker's `stats socket` or via master process's `master CLI` socket. I expect both to have similar behavior when `set severity-output number` is set. This behavior not being the same on `master HEAD` is leading to crash on client-native. https://github.com/haproxytech/client-native/issues/100

As per my past experience, I've been experiencing delays on issues on client-native project to get responses, so I decided to post here. Sorry for the duplicate, issue on that repo is mostly to fix the nil pointer deferencing crash, and the issue here is around HAproxy behavior. 

### Expected Behavior

Both worker and master socket should have similar behavior. 

### Steps to Reproduce the Behavior

1. Use config provided on Github issue.

2. Execute command on `worker socket`. 

```
$ echo "set severity-output number; prepare acl #0" | socat stdio  /tmp/worker.sock

[6]: New version created: 1
```

3. Execute command on `master socket`. 
```
$ echo "@1 set severity-output number; @1 prepare acl #0" | socat stdio  /tmp/master.sock

New version created: 2
```

### What is your configuration?

```haproxy
global
  stats socket /tmp/worker.sock mode 770 level admin

frontend stats
  mode http
  bind *:8081
  stats enable
  stats uri /stats
  stats refresh 1s

frontend http
  mode http
  bind :::8080
  acl is_offline_upstream var(sess.src_l4_ip) -m ip -f /tmp/offline_upstream.lst
  default_backend default_backend

backend default_backend
    balance roundrobin
    mode http
    server rgw1 localhost:8090 check
```
```


### Output of `haproxy -vv`

```plain
$ ./haproxy -vv
HAProxy version 2.9-dev9-f1f8e2-108 2023/11/16 - https://haproxy.org/
Status: development branch - not safe for use in production.
Known bugs: https://github.com/haproxy/haproxy/issues?q=is:issue+is:open
Running on: Linux 6.3.0-rc6+ #9 SMP PREEMPT_DYNAMIC Sun May 14 13:07:46 PDT 2023 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-atomic-alignment
  OPTIONS = USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_SYSTEMD=1 USE_STATIC_PCRE=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT +PCRE -PCRE2 -PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL -PROMEX -PTHREAD_EMULATION -QUIC -QUIC_OPENSSL_COMPAT +RT +SHM_OPEN -SLZ +SSL +STATIC_PCRE -STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL +ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=64).
Built with OpenSSL version : OpenSSL 3.0.2 15 Mar 2022
Running on OpenSSL version : OpenSSL 3.0.2 15 Mar 2022
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
OpenSSL providers loaded : default
Built with Lua version : Lua 5.3.6
Built with network namespace support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE version : 8.39 2016-06-14
Running on PCRE version : 8.39 2016-06-14
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 11.4.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : none

Available filters :
        [BWLIM] bwlim-in
        [BWLIM] bwlim-out
        [CACHE] cache
        [COMP] compression
        [FCGI] fcgi-app
        [SPOE] spoe
        [TRACE] trace

shadyabhi@dev-homelab:~/playground/haproxy$
```
```


### Last Outputs and Backtraces

_No response_

### Additional Information

_No response_

**Comment by wlallemand** [comment]
Hello,

When sending requests from the master CLI to a worker, the commands are sent separately in new CLI sessions. This is how it was designed at the beginning because you can chain commands for multiple workers.

We had to handle special cases for "expert-mode" and "experimental-mode" which have the same problem.
I'll had an exception for this case as well.

**Comment by shadyabhi** [comment]
Hi @wlallemand , thank you for the response. Can we expect to have this go in next release? This is basically avoiding the use of atomic operations for ACL when `MasterSocket` is used.  https://github.com/haproxytech/client-native/issues/100

In our environment, we already had a use-case for atomic operations, but it was working so far as we used the stats socket, not master-cli socket. However, with the migration to master CLI socket, we've hit this blocker.

**Comment by wlallemand** [comment]
I'll try to make the fix so it is in the next subversion, it's a bug so It will be backported everywhere anyway.

**Comment by wlallemand** [comment]
@shadyabhi fyi this was released in 2.9.1.
